### PR TITLE
Support for Custom Annotations on Linkerd-Viz Prometheus Deployment

### LINKED ISSUE #11365: Support for Custom Annotations on Linkerd-Viz Prometheus Deployment
### What problem are you trying to solve?

It would be helpful to be able to specify additional custom Kubernetes annotations for the prometheus deployment within the linkerd-viz helm chart.

Currently, there is no direct way to do this (i.e. can't use `--set` or a `values.yaml`).

### How should the problem be solved?

Just add a small include on the annotations spec for the prometheus deployment with a default empty object on the `values.yaml` that can be overwritten.

### Any alternatives you've considered?

Only forking this repo to make the change.

### How would users interact with this feature?

By default nothing will change for users, but if they so wish they can specify additional annotations via a `--set` or `values.yaml` file.

### Would you like to work on this feature?

maybe

### PR DESCRIPTION
**Pull request by cemenson**
Support for Custom Annotations on Linkerd-Viz Prometheus Deployment

Currently, adding custom annotations to the prometheus deployment isn't supported. This can't be done either with `--set` helm flag or by adding relevant config to `values.yaml`.

By adding additional helm values by way of a `with` keyword within the prometheus deployment yaml, we can support custom annotations. Setting the value of the reverenced value (`.Values.prometheusAnnotations`) to empty will ensure this doesn't break or modify any existing deployments.

Once merged, setting `prometheusAnnotations` in the `values.yaml` or via a `--set` helm flag will result in specified annotations being included in the prometheus deployment and pod spec.

Fixes #11365

I agree to the DCO for all the commits in this PR.

### CODE REVIEW

File: viz/charts/linkerd-viz/values.yaml [hunk 1]
```diff
@@ -64,6 +64,9 @@ enablePSP: false
 # -- url of external prometheus instance
 prometheusUrl: ""
 
+# -- annotations for the prometheus deployment
+prometheusAnnotations: {}
```
Comment (alpeb) [inline viz/charts/linkerd-viz/values.yaml:68]: Can you please add this under the `prometheus` root below instead?

**Comment by cemenson** [comment]
@alpeb is this good to merge?

**Comment by cemenson** [comment]
@adleong could I trouble you for a review of this?

**Comment by adleong** [comment]
Hi @cemenson, can you tell me more about the use case?  What kinds of annotations do you need to add to the Prometheus deployment and why?  I'll note that we already have a `podAnnotations` value which can be used to set annotations on the pods themselves.  It would be great to understand in what scenarios annotations need to be added to the deployement vs added to the pod template.

**Comment by cemenson** [comment]
> Hi @cemenson, can you tell me more about the use case? What kinds of annotations do you need to add to the Prometheus deployment and why? I'll note that we already have a `podAnnotations` value which can be used to set annotations on the pods themselves. It would be great to understand in what scenarios annotations need to be added to the deployement vs added to the pod template.

Hi @adleong. For our specific case, we need to set an annotation `config.linkerd.io/skip-inbound-ports`. There is no way to do this as part of the helm install, other than as you said setting the `podAnnotations` value. Whilst this would work, it would also apply (as I understand from the chart) the annotation to all the other pods that are part of the linkerd-viz chart (`tap`, `web`, etc.), which is not desired.

**Comment by adleong** [comment]
thanks @cemenson, that context is super helpful!

How would you feel about renaming the value to `prometheus.podAnnotations` and having it apply just to the prometheus pod template and not the deployment itself?

**Comment by cemenson** [comment]
@adleong yes that's fine, I'll make those changes.

**Comment by cemenson** [comment]
> thanks @cemenson, that context is super helpful!
> 
> How would you feel about renaming the value to `prometheus.podAnnotations` and having it apply just to the prometheus pod template and not the deployment itself?

Yes that's fine @adleong. I've made those changes now.

**Comment by adleong** [comment]
Thanks, @cemenson.  I'm seeing some weird CI failures that I can't reproduce locally.  Could you try pulling the latest main and merging it into this branch?

**Review by adleong — APPROVED** [review]

**Comment by adleong** [comment]
🎉 🎉 🎉

**Pull request merged**